pkgs/development/libraries/boost/generic.nix

```diff
@@ -149,7 +149,7 @@ stdenv.mkDerivation {
   configureScript = "./bootstrap.sh";
   configureFlags = commonConfigureFlags ++ [
     "--with-icu=${icu}"
-    "--with-python=${python}/bin/python"
+    "--with-python=${python}/bin/python${python.majorVersion}"
```

Comment:
I suppose that makes some sense. Eyeballing the python expressions, it looks like they all have an `interpreter` pass-through attribute which is set to suitable values. Could you just use `--with-python=${python.interpreter}`, perhaps? It would still cause a boost rebuild, but would feel less "implementation detailey".